src/contexts/LoginContext.tsx

```diff
@@ -69,7 +72,11 @@ const LoginProvider = ({
       return response
     },
     async (error) => {
-      if (error.response && error.response.status === 401) {
+      if (
+        error.response &&
+        error.response.status === 401 &&
+        !LOGIN_PATHS.includes(pathname)
```

Comment:
actually on reading this, it looks very suspicous. if i'm on `/` or `/sgid-callback` and i get a `401 unauthorized`, i'm not logged out because there might be another request in flight to sgid?

Reply:
My guess is that the unhappy flow on `/sgid-callback` is - 
1. `verify-login` is called, and `whoami` is called, since it's part of the login context
2. `whoami` resolves with 401 (expected). This triggers `logout`
3. `verify-login` resolves. The cookie with the user data is set
4. `logout` resolves. The cookie with user data is removed
5. User gets redirected to `sites`. They no longer have a valid session cookie, so they get logged out when `whoami` triggers again

I think users who aren't affected by this just happen to have steps 3 and 4 swapped.

Logically i don't think it makes sense to call `logout` from the login pages anyway, since logged in users should be redirected away anyway, and logged out users are already logged out

Reply:
I think this might be possible! I'm opting for the quickest solution here to test it out though, since we can't be sure that this is the actual source of the issue